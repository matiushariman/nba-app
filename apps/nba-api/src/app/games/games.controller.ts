import { Controller, Get, Param } from '@nestjs/common';

import { GamesService } from './games.service';

import type { GetGameByIdReq } from '@nba-app/types-nba';

@Controller('games')
export class GamesController {
  constructor(private readonly gamesService: GamesService) {}

  @Get()
  getGames() {
    return this.gamesService.getGames();
  }

  @Get(':gameId')
  getGameById(@Param() params: GetGameByIdReq) {
    return this.gamesService.getGameById(params.gameId);
  }
}
