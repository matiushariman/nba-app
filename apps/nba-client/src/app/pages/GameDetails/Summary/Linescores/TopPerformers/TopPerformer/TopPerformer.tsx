import type { TopPerformerProps } from './TopPerformer.types';

import TopPerformerStats from './TopPerformerStats';
import PlayerHeadshot from '../../../../../../components/PlayerHeadshot';

const TopPerformer = ({
  personId,
  name,
  teamTricode,
  jerseyNum,
  position,
  points,
  rebounds,
  assists,
}: TopPerformerProps) => (
  <div className="flex items-center">
    <div className="flex flex-col items-center gap-1 w-[115px]">
      <div>
        <PlayerHeadshot personId={personId} name={'name'} size="md" />
      </div>
      <div>
        <p className="text-sm text-center">{name}</p>
        <p className="text-xs text-gray-500 text-center">{`${teamTricode} | #${jerseyNum} | ${position}`}</p>
      </div>
    </div>
    <div className="grid grid-cols-3 flex-grow">
      <div>
        <TopPerformerStats label="Pts" value={points} />
      </div>
      <div>
        <TopPerformerStats label="Reb" value={rebounds} />
      </div>
      <div>
        <TopPerformerStats label="Ast" value={assists} />
      </div>
    </div>
  </div>
);

export default TopPerformer;
