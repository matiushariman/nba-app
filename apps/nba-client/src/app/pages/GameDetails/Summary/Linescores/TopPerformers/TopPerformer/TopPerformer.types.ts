import type { Leaders } from '@nba-app/types-nba';

export type TopPerformerProps = Pick<
  Leaders,
  | 'personId'
  | 'teamTricode'
  | 'name'
  | 'jerseyNum'
  | 'position'
  | 'points'
  | 'rebounds'
  | 'assists'
>;
