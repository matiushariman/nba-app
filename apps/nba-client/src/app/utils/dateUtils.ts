import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import timezone from 'dayjs/plugin/timezone';

dayjs.extend(utc);
dayjs.extend(timezone);

interface ConvertDateToEtParams {
  date?: string | number | dayjs.Dayjs | Date;
  format?: string;
}

export const convertDateToEt = ({
  date,
  format = 'MMMM D, YYYY h:mm A',
}: ConvertDateToEtParams) => dayjs(date).tz('America/Toronto').format(format);
