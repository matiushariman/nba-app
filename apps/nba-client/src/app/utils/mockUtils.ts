export const shouldUseMock = process.env['NODE_ENV'] === 'development';
