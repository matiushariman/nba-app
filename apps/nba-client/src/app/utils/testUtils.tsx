import { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from 'react-query';

import '../../styles.css';

const queryClient = new QueryClient();

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => (
  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';

export { customRender as render };
