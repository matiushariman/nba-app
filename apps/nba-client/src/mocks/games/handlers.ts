import { rest } from 'msw';
import { GetGamesRes, GetGameByIdRes } from '@nba-app/types-nba';

export const mockGames = {
  meta: {
    version: 1,
    request:
      'https://nba-prod-us-east-1-mediaops-stats.s3.amazonaws.com/NBA/liveData/scoreboard/todaysScoreboard_00.json',
    time: '2022-12-07 09:38:05.385',
    code: 200,
  },
  scoreboard: {
    gameDate: '2022-12-07',
    leagueId: '00',
    leagueName: 'National Basketball Association',
    games: [
      {
        gameId: '0022200370',
        gameCode: '20221207/INDMIN',
        gameStatus: 2,
        gameStatusText: 'Q3 09:20',
        period: 3,
        gameClock: 'PT09M20.00S',
        gameTimeUTC: '2022-12-08T01:00:00Z',
        gameEt: '2022-12-07T20:00:00-05:00',
        regulationPeriods: 4,
        ifNecessary: false,
        seriesGameNumber: '',
        seriesText: '',
        homeTeam: {
          teamId: 1610612750,
          teamName: 'Timberwolves',
          teamCity: 'Minnesota',
          teamTricode: 'MIN',
          wins: 11,
          losses: 12,
          score: 65,
          seed: null,
          inBonus: '0',
          timeoutsRemaining: 5,
          periods: [
            {
              period: 1,
              periodType: 'REGULAR',
              score: 35,
            },
            {
              period: 2,
              periodType: 'REGULAR',
              score: 28,
            },
            {
              period: 3,
              periodType: 'REGULAR',
              score: 2,
            },
            {
              period: 4,
              periodType: 'REGULAR',
              score: 0,
            },
          ],
        },
        awayTeam: {
          teamId: 1610612754,
          teamName: 'Pacers',
          teamCity: 'Indiana',
          teamTricode: 'IND',
          wins: 13,
          losses: 11,
          score: 67,
          seed: null,
          inBonus: '0',
          timeoutsRemaining: 4,
          periods: [
            {
              period: 1,
              periodType: 'REGULAR',
              score: 17,
            },
            {
              period: 2,
              periodType: 'REGULAR',
              score: 44,
            },
            {
              period: 3,
              periodType: 'REGULAR',
              score: 6,
            },
            {
              period: 4,
              periodType: 'REGULAR',
              score: 0,
            },
          ],
        },
        gameLeaders: {
          homeLeaders: {
            personId: 1630162,
            name: 'Anthony Edwards',
            jerseyNum: '1',
            position: 'G',
            teamTricode: 'MIN',
            playerSlug: 'anthony-edwards',
            points: 16,
            rebounds: 5,
            assists: 5,
          },
          awayLeaders: {
            personId: 1630169,
            name: 'Tyrese Haliburton',
            jerseyNum: '0',
            position: 'G',
            teamTricode: 'IND',
            playerSlug: 'tyrese-haliburton',
            points: 13,
            rebounds: 2,
            assists: 8,
          },
        },
        pbOdds: {
          team: null,
          odds: 0,
          suspended: 0,
        },
      },
      {
        gameId: '0022200371',
        gameCode: '20221207/DETNOP',
        gameStatus: 2,
        gameStatusText: 'Q3 06:00',
        period: 3,
        gameClock: 'PT06M00.00S',
        gameTimeUTC: '2022-12-08T01:00:00Z',
        gameEt: '2022-12-07T20:00:00-05:00',
        regulationPeriods: 4,
        ifNecessary: false,
        seriesGameNumber: '',
        seriesText: '',
        homeTeam: {
          teamId: 1610612740,
          teamName: 'Pelicans',
          teamCity: 'New Orleans',
          teamTricode: 'NOP',
          wins: 15,
          losses: 8,
          score: 67,
          seed: null,
          inBonus: '0',
          timeoutsRemaining: 3,
          periods: [
            {
              period: 1,
              periodType: 'REGULAR',
              score: 27,
            },
            {
              period: 2,
              periodType: 'REGULAR',
              score: 29,
            },
            {
              period: 3,
              periodType: 'REGULAR',
              score: 11,
            },
            {
              period: 4,
              periodType: 'REGULAR',
              score: 0,
            },
          ],
        },
        awayTeam: {
          teamId: 1610612765,
          teamName: 'Pistons',
          teamCity: 'Detroit',
          teamTricode: 'DET',
          wins: 7,
          losses: 19,
          score: 57,
          seed: null,
          inBonus: '0',
          timeoutsRemaining: 5,
          periods: [
            {
              period: 1,
              periodType: 'REGULAR',
              score: 28,
            },
            {
              period: 2,
              periodType: 'REGULAR',
              score: 15,
            },
            {
              period: 3,
              periodType: 'REGULAR',
              score: 14,
            },
            {
              period: 4,
              periodType: 'REGULAR',
              score: 0,
            },
          ],
        },
        gameLeaders: {
          homeLeaders: {
            personId: 1629627,
            name: 'Zion Williamson',
            jerseyNum: '1',
            position: 'F',
            teamTricode: 'NOP',
            playerSlug: 'zion-williamson',
            points: 22,
            rebounds: 8,
            assists: 3,
          },
          awayLeaders: {
            personId: 1630165,
            name: 'Killian Hayes',
            jerseyNum: '7',
            position: 'G',
            teamTricode: 'DET',
            playerSlug: 'killian-hayes',
            points: 14,
            rebounds: 5,
            assists: 6,
          },
        },
        pbOdds: {
          team: null,
          odds: 0,
          suspended: 0,
        },
      },
      {
        gameId: '0022200368',
        gameCode: '20221207/OKCMEM',
        gameStatus: 2,
        gameStatusText: 'Q3 04:17',
        period: 3,
        gameClock: 'PT04M17.00S',
        gameTimeUTC: '2022-12-08T01:00:00Z',
        gameEt: '2022-12-07T20:00:00-05:00',
        regulationPeriods: 4,
        ifNecessary: false,
        seriesGameNumber: '',
        seriesText: '',
        homeTeam: {
          teamId: 1610612763,
          teamName: 'Grizzlies',
          teamCity: 'Memphis',
          teamTricode: 'MEM',
          wins: 15,
          losses: 9,
          score: 82,
          seed: null,
          inBonus: '0',
          timeoutsRemaining: 4,
          periods: [
            {
              period: 1,
              periodType: 'REGULAR',
              score: 27,
            },
            {
              period: 2,
              periodType: 'REGULAR',
              score: 30,
            },
            {
              period: 3,
              periodType: 'REGULAR',
              score: 25,
            },
            {
              period: 4,
              periodType: 'REGULAR',
              score: 0,
            },
          ],
        },
        awayTeam: {
          teamId: 1610612760,
          teamName: 'Thunder',
          teamCity: 'Oklahoma City',
          teamTricode: 'OKC',
          wins: 11,
          losses: 13,
          score: 73,
          seed: null,
          inBonus: '0',
          timeoutsRemaining: 4,
          periods: [
            {
              period: 1,
              periodType: 'REGULAR',
              score: 25,
            },
            {
              period: 2,
              periodType: 'REGULAR',
              score: 27,
            },
            {
              period: 3,
              periodType: 'REGULAR',
              score: 21,
            },
            {
              period: 4,
              periodType: 'REGULAR',
              score: 0,
            },
          ],
        },
        gameLeaders: {
          homeLeaders: {
            personId: 1629630,
            name: 'Ja Morant',
            jerseyNum: '12',
            position: 'G',
            teamTricode: 'MEM',
            playerSlug: 'ja-morant',
            points: 21,
            rebounds: 12,
            assists: 8,
          },
          awayLeaders: {
            personId: 1628983,
            name: 'Shai Gilgeous-Alexander',
            jerseyNum: '2',
            position: 'GF',
            teamTricode: 'OKC',
            playerSlug: 'shai-gilgeous-alexander',
            points: 19,
            rebounds: 3,
            assists: 5,
          },
        },
        pbOdds: {
          team: null,
          odds: 0,
          suspended: 0,
        },
      },
      {
        gameId: '0022200364',
        gameCode: '20221207/CHABKN',
        gameStatus: 2,
        gameStatusText: 'Q4 03:22',
        period: 4,
        gameClock: 'PT03M22.00S',
        gameTimeUTC: '2022-12-08T00:30:00Z',
        gameEt: '2022-12-07T19:30:00-05:00',
        regulationPeriods: 4,
        ifNecessary: false,
        seriesGameNumber: '',
        seriesText: '',
        homeTeam: {
          teamId: 1610612751,
          teamName: 'Nets',
          teamCity: 'Brooklyn',
          teamTricode: 'BKN',
          wins: 13,
          losses: 12,
          score: 113,
          seed: null,
          inBonus: '0',
          timeoutsRemaining: 2,
          periods: [
            {
              period: 1,
              periodType: 'REGULAR',
              score: 38,
            },
            {
              period: 2,
              periodType: 'REGULAR',
              score: 35,
            },
            {
              period: 3,
              periodType: 'REGULAR',
              score: 23,
            },
            {
              period: 4,
              periodType: 'REGULAR',
              score: 17,
            },
          ],
        },
        awayTeam: {
          teamId: 1610612766,
          teamName: 'Hornets',
          teamCity: 'Charlotte',
          teamTricode: 'CHA',
          wins: 7,
          losses: 17,
          score: 112,
          seed: null,
          inBonus: '0',
          timeoutsRemaining: 3,
          periods: [
            {
              period: 1,
              periodType: 'REGULAR',
              score: 28,
            },
            {
              period: 2,
              periodType: 'REGULAR',
              score: 27,
            },
            {
              period: 3,
              periodType: 'REGULAR',
              score: 32,
            },
            {
              period: 4,
              periodType: 'REGULAR',
              score: 25,
            },
          ],
        },
        gameLeaders: {
          homeLeaders: {
            personId: 202681,
            name: 'Kyrie Irving',
            jerseyNum: '11',
            position: 'G',
            teamTricode: 'BKN',
            playerSlug: 'kyrie-irving',
            points: 29,
            rebounds: 5,
            assists: 7,
          },
          awayLeaders: {
            personId: 1626162,
            name: 'Kelly Oubre Jr.',
            jerseyNum: '12',
            position: 'FG',
            teamTricode: 'CHA',
            playerSlug: 'kelly-oubre-jr',
            points: 28,
            rebounds: 6,
            assists: 1,
          },
        },
        pbOdds: {
          team: null,
          odds: 0,
          suspended: 0,
        },
      },
      {
        gameId: '0022200369',
        gameCode: '20221207/SACMIL',
        gameStatus: 2,
        gameStatusText: 'Q3 02:25',
        period: 3,
        gameClock: 'PT02M25.00S',
        gameTimeUTC: '2022-12-08T01:00:00Z',
        gameEt: '2022-12-07T20:00:00-05:00',
        regulationPeriods: 4,
        ifNecessary: false,
        seriesGameNumber: '',
        seriesText: '',
        homeTeam: {
          teamId: 1610612749,
          teamName: 'Bucks',
          teamCity: 'Milwaukee',
          teamTricode: 'MIL',
          wins: 17,
          losses: 6,
          score: 95,
          seed: null,
          inBonus: '0',
          timeoutsRemaining: 5,
          periods: [
            {
              period: 1,
              periodType: 'REGULAR',
              score: 34,
            },
            {
              period: 2,
              periodType: 'REGULAR',
              score: 34,
            },
            {
              period: 3,
              periodType: 'REGULAR',
              score: 27,
            },
            {
              period: 4,
              periodType: 'REGULAR',
              score: 0,
            },
          ],
        },
        awayTeam: {
          teamId: 1610612758,
          teamName: 'Kings',
          teamCity: 'Sacramento',
          teamTricode: 'SAC',
          wins: 13,
          losses: 9,
          score: 86,
          seed: null,
          inBonus: '0',
          timeoutsRemaining: 3,
          periods: [
            {
              period: 1,
              periodType: 'REGULAR',
              score: 29,
            },
            {
              period: 2,
              periodType: 'REGULAR',
              score: 32,
            },
            {
              period: 3,
              periodType: 'REGULAR',
              score: 25,
            },
            {
              period: 4,
              periodType: 'REGULAR',
              score: 0,
            },
          ],
        },
        gameLeaders: {
          homeLeaders: {
            personId: 201950,
            name: 'Jrue Holiday',
            jerseyNum: '21',
            position: 'G',
            teamTricode: 'MIL',
            playerSlug: 'jrue-holiday',
            points: 31,
            rebounds: 3,
            assists: 5,
          },
          awayLeaders: {
            personId: 1627734,
            name: 'Domantas Sabonis',
            jerseyNum: '10',
            position: 'FC',
            teamTricode: 'SAC',
            playerSlug: 'domantas-sabonis',
            points: 18,
            rebounds: 7,
            assists: 6,
          },
        },
        pbOdds: {
          team: null,
          odds: 0,
          suspended: 0,
        },
      },
      {
        gameId: '0022200372',
        gameCode: '20221207/GSWUTA',
        gameStatus: 2,
        gameStatusText: 'Q2 12:00',
        period: 2,
        gameClock: 'PT12M00.00S',
        gameTimeUTC: '2022-12-08T02:00:00Z',
        gameEt: '2022-12-07T21:00:00-05:00',
        regulationPeriods: 4,
        ifNecessary: false,
        seriesGameNumber: '',
        seriesText: '',
        homeTeam: {
          teamId: 1610612762,
          teamName: 'Jazz',
          teamCity: 'Utah',
          teamTricode: 'UTA',
          wins: 14,
          losses: 12,
          score: 30,
          seed: null,
          inBonus: '0',
          timeoutsRemaining: 6,
          periods: [
            {
              period: 1,
              periodType: 'REGULAR',
              score: 30,
            },
            {
              period: 2,
              periodType: 'REGULAR',
              score: 0,
            },
            {
              period: 3,
              periodType: 'REGULAR',
              score: 0,
            },
            {
              period: 4,
              periodType: 'REGULAR',
              score: 0,
            },
          ],
        },
        awayTeam: {
          teamId: 1610612744,
          teamName: 'Warriors',
          teamCity: 'Golden State',
          teamTricode: 'GSW',
          wins: 13,
          losses: 12,
          score: 35,
          seed: null,
          inBonus: '0',
          timeoutsRemaining: 6,
          periods: [
            {
              period: 1,
              periodType: 'REGULAR',
              score: 35,
            },
            {
              period: 2,
              periodType: 'REGULAR',
              score: 0,
            },
            {
              period: 3,
              periodType: 'REGULAR',
              score: 0,
            },
            {
              period: 4,
              periodType: 'REGULAR',
              score: 0,
            },
          ],
        },
        gameLeaders: {
          homeLeaders: {
            personId: 1627736,
            name: 'Malik Beasley',
            jerseyNum: '5',
            position: 'G',
            teamTricode: 'UTA',
            playerSlug: 'malik-beasley',
            points: 6,
            rebounds: 2,
            assists: 0,
          },
          awayLeaders: {
            personId: 1629673,
            name: 'Jordan Poole',
            jerseyNum: '3',
            position: 'G',
            teamTricode: 'GSW',
            playerSlug: 'jordan-poole',
            points: 13,
            rebounds: 0,
            assists: 4,
          },
        },
        pbOdds: {
          team: null,
          odds: 0,
          suspended: 0,
        },
      },
      {
        gameId: '0022200365',
        gameCode: '20221207/ATLNYK',
        gameStatus: 2,
        gameStatusText: 'Q4 11:44',
        period: 4,
        gameClock: 'PT11M44.00S',
        gameTimeUTC: '2022-12-08T00:30:00Z',
        gameEt: '2022-12-07T19:30:00-05:00',
        regulationPeriods: 4,
        ifNecessary: false,
        seriesGameNumber: '',
        seriesText: '',
        homeTeam: {
          teamId: 1610612752,
          teamName: 'Knicks',
          teamCity: 'New York',
          teamTricode: 'NYK',
          wins: 11,
          losses: 13,
          score: 89,
          seed: null,
          inBonus: '0',
          timeoutsRemaining: 4,
          periods: [
            {
              period: 1,
              periodType: 'REGULAR',
              score: 31,
            },
            {
              period: 2,
              periodType: 'REGULAR',
              score: 22,
            },
            {
              period: 3,
              periodType: 'REGULAR',
              score: 36,
            },
            {
              period: 4,
              periodType: 'REGULAR',
              score: 0,
            },
          ],
        },
        awayTeam: {
          teamId: 1610612737,
          teamName: 'Hawks',
          teamCity: 'Atlanta',
          teamTricode: 'ATL',
          wins: 13,
          losses: 11,
          score: 69,
          seed: null,
          inBonus: '0',
          timeoutsRemaining: 3,
          periods: [
            {
              period: 1,
              periodType: 'REGULAR',
              score: 20,
            },
            {
              period: 2,
              periodType: 'REGULAR',
              score: 21,
            },
            {
              period: 3,
              periodType: 'REGULAR',
              score: 26,
            },
            {
              period: 4,
              periodType: 'REGULAR',
              score: 2,
            },
          ],
        },
        gameLeaders: {
          homeLeaders: {
            personId: 203944,
            name: 'Julius Randle',
            jerseyNum: '30',
            position: 'F',
            teamTricode: 'NYK',
            playerSlug: 'julius-randle',
            points: 32,
            rebounds: 14,
            assists: 5,
          },
          awayLeaders: {
            personId: 203991,
            name: 'Clint Capela',
            jerseyNum: '15',
            position: 'F',
            teamTricode: 'ATL',
            playerSlug: 'clint-capela',
            points: 13,
            rebounds: 11,
            assists: 0,
          },
        },
        pbOdds: {
          team: null,
          odds: 0,
          suspended: 0,
        },
      },
      {
        gameId: '0022200367',
        gameCode: '20221207/WASCHI',
        gameStatus: 2,
        gameStatusText: 'Q3 05:19',
        period: 3,
        gameClock: 'PT05M19.00S',
        gameTimeUTC: '2022-12-08T01:00:00Z',
        gameEt: '2022-12-07T20:00:00-05:00',
        regulationPeriods: 4,
        ifNecessary: false,
        seriesGameNumber: '',
        seriesText: '',
        homeTeam: {
          teamId: 1610612741,
          teamName: 'Bulls',
          teamCity: 'Chicago',
          teamTricode: 'CHI',
          wins: 9,
          losses: 14,
          score: 70,
          seed: null,
          inBonus: '0',
          timeoutsRemaining: 4,
          periods: [
            {
              period: 1,
              periodType: 'REGULAR',
              score: 29,
            },
            {
              period: 2,
              periodType: 'REGULAR',
              score: 23,
            },
            {
              period: 3,
              periodType: 'REGULAR',
              score: 18,
            },
            {
              period: 4,
              periodType: 'REGULAR',
              score: 0,
            },
          ],
        },
        awayTeam: {
          teamId: 1610612764,
          teamName: 'Wizards',
          teamCity: 'Washington',
          teamTricode: 'WAS',
          wins: 11,
          losses: 13,
          score: 65,
          seed: null,
          inBonus: '1',
          timeoutsRemaining: 4,
          periods: [
            {
              period: 1,
              periodType: 'REGULAR',
              score: 22,
            },
            {
              period: 2,
              periodType: 'REGULAR',
              score: 27,
            },
            {
              period: 3,
              periodType: 'REGULAR',
              score: 16,
            },
            {
              period: 4,
              periodType: 'REGULAR',
              score: 0,
            },
          ],
        },
        gameLeaders: {
          homeLeaders: {
            personId: 202696,
            name: 'Nikola Vucevic',
            jerseyNum: '9',
            position: 'F',
            teamTricode: 'CHI',
            playerSlug: 'nikola-vucevic',
            points: 18,
            rebounds: 8,
            assists: 3,
          },
          awayLeaders: {
            personId: 1630166,
            name: 'Deni Avdija',
            jerseyNum: '9',
            position: 'F',
            teamTricode: 'WAS',
            playerSlug: 'deni-avdija',
            points: 8,
            rebounds: 5,
            assists: 3,
          },
        },
        pbOdds: {
          team: null,
          odds: 0,
          suspended: 0,
        },
      },
      {
        gameId: '0022200363',
        gameCode: '20221207/LACORL',
        gameStatus: 2,
        gameStatusText: 'OT 02:23',
        period: 5,
        gameClock: 'PT02M23.00S',
        gameTimeUTC: '2022-12-08T00:00:00Z',
        gameEt: '2022-12-07T19:00:00-05:00',
        regulationPeriods: 4,
        ifNecessary: false,
        seriesGameNumber: '',
        seriesText: '',
        homeTeam: {
          teamId: 1610612753,
          teamName: 'Magic',
          teamCity: 'Orlando',
          teamTricode: 'ORL',
          wins: 5,
          losses: 20,
          score: 106,
          seed: null,
          inBonus: '0',
          timeoutsRemaining: 2,
          periods: [
            {
              period: 1,
              periodType: 'REGULAR',
              score: 14,
            },
            {
              period: 2,
              periodType: 'REGULAR',
              score: 26,
            },
            {
              period: 3,
              periodType: 'REGULAR',
              score: 31,
            },
            {
              period: 4,
              periodType: 'REGULAR',
              score: 28,
            },
            {
              period: 5,
              periodType: 'OVERTIME',
              score: 7,
            },
          ],
        },
        awayTeam: {
          teamId: 1610612746,
          teamName: 'Clippers',
          teamCity: 'LA',
          teamTricode: 'LAC',
          wins: 14,
          losses: 11,
          score: 105,
          seed: null,
          inBonus: '0',
          timeoutsRemaining: 2,
          periods: [
            {
              period: 1,
              periodType: 'REGULAR',
              score: 32,
            },
            {
              period: 2,
              periodType: 'REGULAR',
              score: 17,
            },
            {
              period: 3,
              periodType: 'REGULAR',
              score: 29,
            },
            {
              period: 4,
              periodType: 'REGULAR',
              score: 21,
            },
            {
              period: 5,
              periodType: 'OVERTIME',
              score: 6,
            },
          ],
        },
        gameLeaders: {
          homeLeaders: {
            personId: 1629021,
            name: 'Moritz Wagner',
            jerseyNum: '21',
            position: 'FC',
            teamTricode: 'ORL',
            playerSlug: 'moritz-wagner',
            points: 20,
            rebounds: 12,
            assists: 3,
          },
          awayLeaders: {
            personId: 1629611,
            name: 'Terance Mann',
            jerseyNum: '14',
            position: 'GF',
            teamTricode: 'LAC',
            playerSlug: 'terance-mann',
            points: 19,
            rebounds: 9,
            assists: 4,
          },
        },
        pbOdds: {
          team: null,
          odds: 0,
          suspended: 0,
        },
      },
      {
        gameId: '0022200366',
        gameCode: '20221207/LALTOR',
        gameStatus: 2,
        gameStatusText: 'Q4 07:14',
        period: 4,
        gameClock: 'PT07M14.00S',
        gameTimeUTC: '2022-12-08T00:30:00Z',
        gameEt: '2022-12-07T19:30:00-05:00',
        regulationPeriods: 4,
        ifNecessary: false,
        seriesGameNumber: '',
        seriesText: '',
        homeTeam: {
          teamId: 1610612761,
          teamName: 'Raptors',
          teamCity: 'Toronto',
          teamTricode: 'TOR',
          wins: 12,
          losses: 12,
          score: 110,
          seed: null,
          inBonus: '0',
          timeoutsRemaining: 4,
          periods: [
            {
              period: 1,
              periodType: 'REGULAR',
              score: 39,
            },
            {
              period: 2,
              periodType: 'REGULAR',
              score: 31,
            },
            {
              period: 3,
              periodType: 'REGULAR',
              score: 27,
            },
            {
              period: 4,
              periodType: 'REGULAR',
              score: 13,
            },
          ],
        },
        awayTeam: {
          teamId: 1610612747,
          teamName: 'Lakers',
          teamCity: 'Los Angeles',
          teamTricode: 'LAL',
          wins: 10,
          losses: 13,
          score: 87,
          seed: null,
          inBonus: '0',
          timeoutsRemaining: 2,
          periods: [
            {
              period: 1,
              periodType: 'REGULAR',
              score: 31,
            },
            {
              period: 2,
              periodType: 'REGULAR',
              score: 16,
            },
            {
              period: 3,
              periodType: 'REGULAR',
              score: 28,
            },
            {
              period: 4,
              periodType: 'REGULAR',
              score: 12,
            },
          ],
        },
        gameLeaders: {
          homeLeaders: {
            personId: 1627783,
            name: 'Pascal Siakam',
            jerseyNum: '43',
            position: 'F',
            teamTricode: 'TOR',
            playerSlug: 'pascal-siakam',
            points: 23,
            rebounds: 10,
            assists: 7,
          },
          awayLeaders: {
            personId: 1628418,
            name: 'Thomas Bryant',
            jerseyNum: '31',
            position: 'FC',
            teamTricode: 'LAL',
            playerSlug: 'thomas-bryant',
            points: 12,
            rebounds: 5,
            assists: 0,
          },
        },
        pbOdds: {
          team: null,
          odds: 0,
          suspended: 0,
        },
      },
      {
        gameId: '0022200373',
        gameCode: '20221207/BOSPHX',
        gameStatus: 1,
        gameStatusText: '10:00 pm ET',
        period: 0,
        gameClock: '',
        gameTimeUTC: '2022-12-08T03:00:00Z',
        gameEt: '2022-12-07T22:00:00Z',
        regulationPeriods: 4,
        ifNecessary: false,
        seriesGameNumber: '',
        seriesText: '',
        homeTeam: {
          teamId: 1610612756,
          teamName: 'Suns',
          teamCity: 'Phoenix',
          teamTricode: 'PHX',
          wins: 16,
          losses: 8,
          score: 0,
          seed: null,
          inBonus: null,
          timeoutsRemaining: 0,
          periods: [
            {
              period: 1,
              periodType: 'REGULAR',
              score: 0,
            },
            {
              period: 2,
              periodType: 'REGULAR',
              score: 0,
            },
            {
              period: 3,
              periodType: 'REGULAR',
              score: 0,
            },
            {
              period: 4,
              periodType: 'REGULAR',
              score: 0,
            },
          ],
        },
        awayTeam: {
          teamId: 1610612738,
          teamName: 'Celtics',
          teamCity: 'Boston',
          teamTricode: 'BOS',
          wins: 20,
          losses: 5,
          score: 0,
          seed: null,
          inBonus: null,
          timeoutsRemaining: 0,
          periods: [
            {
              period: 1,
              periodType: 'REGULAR',
              score: 0,
            },
            {
              period: 2,
              periodType: 'REGULAR',
              score: 0,
            },
            {
              period: 3,
              periodType: 'REGULAR',
              score: 0,
            },
            {
              period: 4,
              periodType: 'REGULAR',
              score: 0,
            },
          ],
        },
        gameLeaders: {
          homeLeaders: {
            personId: 0,
            name: '',
            jerseyNum: '',
            position: '',
            teamTricode: 'PHX',
            playerSlug: null,
            points: 0,
            rebounds: 0,
            assists: 0,
          },
          awayLeaders: {
            personId: 0,
            name: '',
            jerseyNum: '',
            position: '',
            teamTricode: 'BOS',
            playerSlug: null,
            points: 0,
            rebounds: 0,
            assists: 0,
          },
        },
        pbOdds: {
          team: null,
          odds: 0,
          suspended: 0,
        },
      },
    ],
  },
};

export const gamesHandlers = [
  rest.get<GetGameByIdRes>('/api/games/:gameId', (req, res, ctx) => {
    const { gameId } = req.params;
    const game = mockGames.scoreboard.games.find(
      (findGame) => findGame.gameId === gameId
    );
    return res(ctx.json(game));
  }),
  rest.get<GetGamesRes>('/api/games', (req, res, ctx) => {
    return res(ctx.json(mockGames));
  }),
];
