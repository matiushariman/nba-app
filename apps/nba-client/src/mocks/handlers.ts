import { boxscoreHandlers } from './boxscore/handlers';
import { gamesHandlers } from './games/handlers';

export const handlers = [...boxscoreHandlers, ...gamesHandlers];
