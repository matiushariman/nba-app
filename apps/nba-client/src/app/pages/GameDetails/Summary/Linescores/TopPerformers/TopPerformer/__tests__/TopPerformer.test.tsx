import TopPerformer from '..';
import { render } from '../../../../../../../utils/testUtils';
import { mockGames } from '../../../../../../../../mocks/games/handlers';

const { awayLeaders } = mockGames.scoreboard.games[0].gameLeaders;

describe('pages/GameDetails/Summary/Linescores/TopPerformers/TopPerformer', () => {
  it('should render TopPerformer that matches snapshot', () => {
    const { baseElement } = render(<TopPerformer {...awayLeaders} />);

    expect(baseElement).toMatchSnapshot();
  });
});
