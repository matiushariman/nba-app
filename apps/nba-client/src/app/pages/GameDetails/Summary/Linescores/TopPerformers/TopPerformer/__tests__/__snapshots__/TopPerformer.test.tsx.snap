// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/GameDetails/Summary/Linescores/TopPerformers/TopPerformer should render TopPerformer that matches snapshot 1`] = `
<body>
  <div>
    <div
      class="flex items-center"
    >
      <div
        class="flex flex-col items-center gap-1 w-[115px]"
      >
        <div>
          <div
            class="w-[48px] h-[48px]"
          >
            <img
              alt="name headshot"
              class="mr-2 bg-gray-300 shadow-sm rounded-full w-full h-full object-cover"
              loading="lazy"
              src="https://cdn.nba.com/headshots/nba/latest/260x190/1629028.png"
            />
          </div>
        </div>
        <div>
          <p
            class="text-sm text-center"
          >
            Deandre Ayton
          </p>
          <p
            class="text-xs text-gray-500 text-center"
          >
            PHX | #22 | C
          </p>
        </div>
      </div>
      <div
        class="grid grid-cols-3 flex-grow"
      >
        <div>
          <div>
            <p
              class="text-center font-bold text-xl"
            >
              28
            </p>
            <p
              class="text-center text-sm uppercase"
            >
              Pts
            </p>
          </div>
        </div>
        <div>
          <div>
            <p
              class="text-center font-bold text-xl"
            >
              12
            </p>
            <p
              class="text-center text-sm uppercase"
            >
              Reb
            </p>
          </div>
        </div>
        <div>
          <div>
            <p
              class="text-center font-bold text-xl"
            >
              5
            </p>
            <p
              class="text-center text-sm uppercase"
            >
              Ast
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
